svd.py

```diff
@@ -1,226 +1,287 @@
+'''Golub-Reinsch SVD algorithm
+     This implementation is based on:
+     "Numerical Recipes in C" section 2.6, 
+     see http://www.grad.hr/nastava/gs/prg/NumericalRecipesinC.pdf
+'''
+
 import numpy as np
 
-# from "Numerical Recipes in C" p. 65
-EPS = 10**-6
+# -------------------------- #
+#
+EPS = 1.e-15  # assumes double precision
```

Comment:
I think it would be better to use the single-precision value, no?

Reply:
I set the EPS to work with double precision to make the code more general. That is, to avoid overflow in the intermediate coputos. In this way, the input matrix A can have a wider range of values.

However, I can set the EPS to work with single precision if you wish. In the same way the algorithm will work on the random data that we are setting in the `test.py`.

Please let me know

Reply:
I assume that this value is larger for a single-precision float, so if the user runs SVD on a single-precision matrix with the double-precision value for EPS, there's still a danger of underflow, but if the user runs SVD on a double-precision matric with the single-precision value, it will still work. Right?

Reply:
SGTM